addons/purchase/models/account_invoice.py

```diff
@@ -73,7 +73,6 @@ def _onchange_purchase_auto_complete(self):
 
         self.purchase_id = False
         self._onchange_currency()
-        self.partner_bank_id = self.bank_partner_id.bank_ids and self.bank_partner_id.bank_ids[0]
```

Comment:
This line has been removed because it sets up a wrong value. Furthermore, the function `_prepare_invoice()` is called earlier and defined the value of `partner_bank_id` which is updated by this function.